src/book.py

```diff
@@ -522,35 +508,54 @@ def _read_kraken_ledgers(self, file_path: Path) -> None:
 
                 row = reader.line_num
 
-                # Skip "duplicate entries" for deposits / withdrawals
-                if _type in dup_state.keys():
-                    skip = dup_state[_type] == dup_skip[_type]
-                    dup_state[_type] = (dup_state[_type] + 1) % 2
-                    if skip:
+                # Skip duplicate entries for deposits / withdrawals and
+                # additional deposit / withdrawals lines for
+                # staking / unstaking / staking reward actions
+                if _type in ["deposit", "withdrawal"]:
+                    if refid not in refids:
+                        refids.append(refid)
                         continue
 
                 # Parse data.
                 utc_time = datetime.datetime.strptime(_utc_time, "%Y-%m-%d %H:%M:%S")
                 utc_time = utc_time.replace(tzinfo=datetime.timezone.utc)
                 change = misc.force_decimal(_amount)
+                # remove the appended .S for staked assets
+                _asset = _asset.replace(".S", "")
```

Comment:
Is this safe or should we change it to `_asset.removesuffix(".S")` (which is actually nothing other than `_asset[:-2] if _asset and _asset.endswith(".S") else _asset`)?